docs/docs/dev_docs/contracts/portals/main.md

```diff
@@ -0,0 +1,331 @@
+
+As this requires logic on the portal itself, it is not something that the protocol can enforce. It must be supported by the application builder when building the portal.
+
+The portal can call the `cancelL2Message` at the `Inbox` when `block.timestamp > deadline` for the message.
+
+```solidity title="IInbox.sol"
+function cancelL2Message(
+    DataStructures.L1ToL2Msg memory _message, 
+    address _feeCollector
+) external returns (bytes32 entryKey);
+```
+
+Building on our token example from earlier, this can be called like:
+
+```solidity title="TokenPortal.sol"
+function cancelL1ToAztecMessage(
+    bytes32 _to,
+    uint256 _amount,
+    uint32 _deadline,
+    bytes32 _secretHash,
+    uint64 _fee
+  ) external returns (bytes32) {
+    IInbox inbox = registry.getInbox();
+    DataStructures.L1Actor memory l1Actor = DataStructures.L1Actor(address(this), block.chainid);
+    DataStructures.L2Actor memory l2Actor = DataStructures.L2Actor(l2TokenAddress, 1);
+    DataStructures.L1ToL2Msg memory message = DataStructures.L1ToL2Msg({
+      sender: l1Actor,
+      recipient: l2Actor,
+      content: Hash.sha256ToField(
+        abi.encodeWithSignature("mint(uint256,bytes32,address)", _amount, _to, msg.sender)
+        ),
+      secretHash: _secretHash,
+      deadline: _deadline,
+      fee: _fee
+    });
+    bytes32 entryKey = inbox.cancelL2Message(message, address(this));
+    // Ensures that `msg.sender == canceller` by using `msg.sender` in the hash computation.
+    underlying.transfer(msg.sender, _amount);
+    return entryKey;
+  }
+```
+
+The example above ensure that the user can cancel his message if it is underpriced.
+
+### Designated caller
+A designated caller is the ability to specify who should be able to call a function that consumes a message. This is useful for ordering of batched messages, as we will see in a second.  
+
+When doing multiple cross-chain calls as one action it is important to consider the order of the calls. Say for example, that you want to do a uniswap trade on L1 because you are a whale and slippage on L2 is too damn high. 
+
+You would practically, withdraw funds from the rollup, swap them on L1, and then deposit the swapped funds back into the rollup. This is a fairly simple process, but it requires that the calls are done in the correct order. For one, if the swap is called before the funds are withdrawn, the swap will fail. And if the deposit is called before the swap, the funds might get lost! 
+
+As the message boxes only will allow the recipient portal to consume the message, we can use this to our advantage to ensure that the calls are done in the correct order. Say that we include a designated "caller" in the messages, and that the portal contract checks that the caller matches the designated caller or designated is address(0) (anyone can call). When the message are to be consumed on L1, it can compute the message as seen below:
```

Comment:
```suggestion
As message boxes only will allow the recipient portal to consume the message, we can use this to our advantage to ensure that the calls are done in the correct order. Say that we include a designated "caller" in the messages, and that the portal contract checks that the caller matches the designated caller or designated is address(0) (anyone can call). When the message are to be consumed on L1, it can compute the message as seen below:
```